src/functions_framework/__init__.py

```diff
@@ -75,30 +97,47 @@ def _is_binary_cloud_event(request):
         and request.headers.get("ce-id")
     )
 
+# TODO(joelgerard): is this really legacy as I have stated?
+def _run_binary_legacy_cloud_event(function, request):
```

Comment:
There are 3 signature types, `http`, `event` (legacy), and `cloudevent`.

The previous implementation of the support for CloudEvents in the `event` signature type shouldn't be there and is outdated.

There's no such thing as a legacy cloudevent. The previous implementation here around `Context` is outdated and can be removed.